### PR TITLE
Make model variables depend on the model

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR changes our model variable names to be dependent on the model. This will allow for automation, but it does put constraints on initialization order.

### CODE REVIEW

File: integration_tests/3dBomex.jl [hunk 1]
```diff
@@ -362,20 +361,20 @@ function ∑tendencies_3d_bomex_gm!(dY, Y, cache, t)
...
-Y = init_state(coords, face_coords, hv_center_space, hv_face_space)
+edmf_cache = get_edmf_cache(grid, hv_center_space, hv_face_space, namelist)
+Y = init_state(edmf_cache.edmf, coords, face_coords, hv_center_space, hv_face_space)
 energy_0 = total_energy(Y)
 mass_0 = sum(Y.cent.ρ) # Computes ∫ρ∂Ω such that quadrature weighting is accounted for.
 theta_0 = sum(Y.cent.ρθ)
 
 # Solve the ODE
 gm_cache = get_gm_cache(Y, coords)
-function get_aux(hv_center_space, hv_face_space, ::Type{FT}; N_up = 1) where {FT}
```
Comment (yairchn) [inline integration_tests/3dBomex.jl:372]: was this n_up =1 added later correctly ?

File: integration_tests/baroclinic_wave.jl [hunk 2]
```diff
@@ -356,20 +355,20 @@ end
 include(joinpath(@__DIR__, "..", "driver", "dycore.jl"))
 include(joinpath(@__DIR__, "..", "driver", "Surface.jl"))
 
-Y = init_state(grid, coords, face_coords, local_geometries)
+edmf_cache = get_edmf_cache(grid, namelist)
+Y = init_state(edmf_cache.edmf, grid, coords, face_coords, local_geometries)
 # Solve the ODE
 gm_cache = get_gm_cache(Y, coords)
-function get_aux(grid; N_up = 1)
```
Comment (yairchn) [inline integration_tests/baroclinic_wave.jl:362]: was this n_up=1 followed later in the changes

File: integration_tests/3dBomex.jl (hunk 1 above)
Comment (charleskawczynski) [reply integration_tests/3dBomex.jl:372]: I hard-coded this for the these mwe cases

**Review by yairchn — APPROVED** [review]
LGTM

**Pull request merged**